packages/common/src/core/schemas/internal/fetchCategoriesUiSchemaElement.ts

```diff
@@ -0,0 +1,66 @@
+import { IArcGISContext } from "../../../ArcGISContext";
+import {
+  IUiSchemaElement,
+  IUiSchemaMessage,
+  UiSchemaMessageTypes,
+} from "../types";
+import { getCategoryItems } from "./getCategoryItems";
+
+/**
+ * Returns the UI schema element needed to render
+ * the categories editing control for an entity.
+ *
+ * @param i18nScope i18n scope for the entity translations
+ * @param entity The entity to build the UI schema for
+ * @returns the UI schema element for thumbnail editing
+ */
+export async function fetchCategoriesUiSchemaElement(
```

Comment:
minor: I might suggest naming this `buildCategoriesUiSchemaElement` or `getCategoriesUiSchemaElement`. In my mind, `fetch...` implies an XHR call

Reply:
Great callout. This was also a point of confusion for me when I started working here, partially because the underlying `getCategoryItems()` function should really be named `fetchCategoryItems()`. I've renamed the helper accordingly so devs know that an XHR is required.